test/Microsoft.Azure.ServiceBus.UnitTests/BrokeredMessageTests.cs

```diff
@@ -105,5 +106,55 @@ void CustomMessageIdGenerator()
 
             BrokeredMessage.SetMessageIdGenerator(null);
         }
+
+        public class When_querying_IsReceived_property
+        {
+            [Fact]
+            [DisplayTestMethodName]
+            void Should_return_false_for_message_that_was_not_sent_and_received()
+            {
+                var message = new BrokeredMessage();
+                message.Properties["dummy"] = "dummy";
+                Assert.False(message.IsReceived);
+            }
+
+            [Theory]
+                [DisplayTestMethodName]
+                [InlineData(ReceiveMode.ReceiveAndDelete)]
+                [InlineData(ReceiveMode.PeekLock)]
+            async Task Should_return_true_for_message_that_was_sent_and_received(ReceiveMode receiveMode)
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), receiveMode);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var receivedMessages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, 1);
+                    Assert.True(receivedMessages.First().IsReceived);
+                }
+                finally
+                {
+                    await queueClient.CloseAsync();
+                }
+            }
+
+            [Fact]
+                [DisplayTestMethodName]
+            async Task Should_return_true_for_peeked_message()
+            {
+                var queueClient = QueueClient.CreateFromConnectionString(TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName), ReceiveMode.PeekLock);
+                try
+                {
+                    await TestUtility.SendMessagesAsync(queueClient.InnerSender, 1);
+                    var peekedMessage = await TestUtility.PeekMessageAsync(queueClient.InnerReceiver);
+                    var result = peekedMessage.IsReceived;
+                    await TestUtility.CompleteMessagesAsync(queueClient.InnerReceiver, new[] { peekedMessage });
```

Comment:
Thinking of it, this test should be changed to test for `false` result _and_ an additional test should be created for testing the property returns `true` after message is completed.